FeedAPIChallenge/Feed API/FeedMapper.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  FeedMapper.swift
+//  FeedAPIChallenge
+//
+//  Created by Ahmed Atef Ali Ahmed on 02.07.21.
+//  Copyright © 2021 Essential Developer Ltd. All rights reserved.
+//
+
+import Foundation
+
+public protocol FeedImageMapper {
+	func map(_ data: Data, response: HTTPURLResponse) -> FeedLoader.Result
+}
+
+public final class RemoteFeedImageMapper: FeedImageMapper {
+	public static let `default` = RemoteFeedImageMapper()
+
+	private static let OK_STATUS_CODE = 200
+
+	public func map(_ data: Data, response: HTTPURLResponse) -> FeedLoader.Result {
+		guard response.statusCode == Self.OK_STATUS_CODE,
+		      let root = try? JSONDecoder().decode(Root.self, from: data)
+		else {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
+		return .success(root.images)
+	}
+}
+
+struct Root: Decodable {
```

Comment:
`Root` is an internal implementation detail of `RemoteFeedImageMapper` thus it should be private, please make it private

```suggestion
private struct Root: Decodable {
```